### ISSUE TITLE
[ColoredText] Proper thread-safety

### ISSUE THREAD

**Issue by jamesqo** [jamesqo/Repository#33]
- Right now, only `getSpans` and `receive` are synchronized. Consider synchronizing other methods that write (or even read) too.

- Is it better to use a `synchronized` block or `synchronized` method? I chose the latter because I'm lazy, but maybe it's better to be explicit about/not leak what you lock on.

**Comment by jamesqo** [comment]
Locking is no longer needed after #61. getSpans always runs after colorWith
